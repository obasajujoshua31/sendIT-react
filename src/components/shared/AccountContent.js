import React, {Component} from "react";
import { withRouter} from 'react-router-dom';

class AccountContent extends Component{
  constructor(props){
    super(props)
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick (){
    this.props.history.push('/signup');
  }
  render(){

  
  return (
    <React.Fragment>
      <div className="service-information">
        <h1><span>Just Call </span>(0908754647846)</h1>
        <p>
          We are always ready for your service
        </p>
        <button onClick ={this.handleClick}>SendIT Today</button>
      </div>
    </React.Fragment>
  );
  }
};

export default withRouter(AccountContent);
