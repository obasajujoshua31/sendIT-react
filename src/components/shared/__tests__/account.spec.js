import React from "react";
import { shallow, mount } from "enzyme";
import sinon from 'sinon';
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import AccountContent, { mapDispatchToProps } from "../AccountContent";

const props = {
  loadParcels: jest.fn(),
  history: {
    push: jest.fn()
  }
};
const initialState = {
  users: {
    singInStatus: "",
    errorMessage: "",
    errorMsg: "",
    signUpStatus: "",
    role: false,
    user: {}
  }
};
const mockStore = configureStore([thunk]);
beforeEach(() => {});
describe("<Account Content> With Connect", () => {
  const store = mockStore(initialState);
  const wrapper = mount(
    <Provider store={store}>
      <Router>
        <AccountContent {...props} />
      </Router>
    </Provider>
  );
  it("should render Account Content Component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
