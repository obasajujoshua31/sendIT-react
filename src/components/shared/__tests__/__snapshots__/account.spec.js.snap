// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account Content> With Connect should render Account Content Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(AccountContent)
        history={
          Object {
            "push": [MockFunction],
          }
        }
        loadParcels={[MockFunction]}
      >
        <Route>
          <AccountContent
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            loadParcels={[MockFunction]}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="service-information"
            >
              <h1>
                <span>
                  Just Call 
                </span>
                (0908754647846)
              </h1>
              <p>
                We are always ready for your service
              </p>
              <button
                onClick={[Function]}
              >
                SendIT Today
              </button>
            </div>
          </AccountContent>
        </Route>
      </withRouter(AccountContent)>
    </Router>
  </BrowserRouter>
</Provider>
`;
